### PR TITLE
add terrafmt to lint in Makefile

### PR DESCRIPTION
**Pull request by danquack**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

In doing some contributory work, I noticed that the pipeline runs terrafmt, but that is not included in the make lint step. In order to cut down on number of checks, this will add it to the local make workflows. Feel free to update or modify if desired.

### CODE REVIEW

**Review by drewmullen — APPROVED** [review]
This worked for me in [PR 22394](https://github.com/hashicorp/terraform-provider-aws/pull/22394/commits/51d7270531307d7b9a1d95a44b1eb43618b22502)
image

**Review by drewmullen — CHANGES_REQUESTED** [review]
can you also include:
`terrafmt fmt ./website --check --pattern '*.markdown'`

this will also check for errors in markdown files

example with `diff` instead of `fmt` to demostrate
```
$ terrafmt diff ./website --pattern '*.markdown'
website/docs/r/vpc_ipam_organization_admin_account.html.markdown:17
 data "aws_caller_identity" "delegated" {
   provider = "awsalternate"
 }
 
 resource "aws_vpc_ipam_organization_admin_account" "example" {
-  delegated_admin_account_id =  data.aws_caller_identity.delegated.account_id
+  delegated_admin_account_id = data.aws_caller_identity.delegated.account_id
 }

**Comment by danquack** [comment]
> can you also include: `terrafmt fmt ./website --check --pattern '*.markdown'`
> 
> this will also check for errors in markdown files
> 
> example with `diff` instead of `fmt` to demostrate
> 
> ```
> $ terrafmt diff ./website --pattern '*.markdown'
> website/docs/r/vpc_ipam_organization_admin_account.html.markdown:17
>  data "aws_caller_identity" "delegated" {
>    provider = "awsalternate"
>  }
>  
>  resource "aws_vpc_ipam_organization_admin_account" "example" {
> -  delegated_admin_account_id =  data.aws_caller_identity.delegated.account_id
> +  delegated_admin_account_id = data.aws_caller_identity.delegated.account_id
>  }
> ```

@drewmullen it looks like this is included in `website-lint-fix` https://github.com/hashicorp/terraform-provider-aws/blob/main/GNUmakefile#L168-L172. Do you just want that added to the lint step instead?

**Comment by danquack** [comment]
> can you also include: `terrafmt fmt ./website --check --pattern '*.markdown'`
> 
> this will also check for errors in markdown files
> 
> example with `diff` instead of `fmt` to demostrate
> 
> ```
> $ terrafmt diff ./website --pattern '*.markdown'
> website/docs/r/vpc_ipam_organization_admin_account.html.markdown:17
>  data "aws_caller_identity" "delegated" {
>    provider = "awsalternate"
>  }
>  
>  resource "aws_vpc_ipam_organization_admin_account" "example" {
> -  delegated_admin_account_id =  data.aws_caller_identity.delegated.account_id
> +  delegated_admin_account_id = data.aws_caller_identity.delegated.account_id
>  }
> ```

@drewmullen it looks like this is included in `website-lint-fix` https://github.com/hashicorp/terraform-provider-aws/blob/main/GNUmakefile#L168-L172. Do you just want that added to the lint rule?

**Comment by drewmullen** [comment]
> @drewmullen it looks like this is included in `website-lint-fix` https://github.com/hashicorp/terraform-provider-aws/blob/main/GNUmakefile#L168-L172. Do you just want that added to the lint rule?

I vote yes; include `website-lint-fix` & `semgrep`. IMO encapsulating phony targets like `lint` aren't useful if they dont help with common problems like these. Open to feedback on this. My guess is they're not included because they rely on docker(?)

**Comment by danquack** [comment]
@drewmullen I added as a separate commit changes that running the new lint steps added. Just a bunch of new lines at the end of files. If you don't want that in this PR, feel free to either remove it or let me know!

File: GNUmakefile [hunk 1]
```diff
@@ -168,11 +178,11 @@ website-lint:
 website-lint-fix:
 	@echo "==> Applying automatic website linter fixes..."
 	@misspell -w -source=text website/
-	@docker run -v $(PWD):/markdown 06kellyjac/markdownlint-cli --fix website/docs/
+	@docker run --platform linux/amd64 -v $(PWD):/markdown 06kellyjac/markdownlint-cli --fix website/docs/
```
Comment (drewmullen) [inline GNUmakefile:181]: context: `--platform linux/amd64` was added due to compatibility issues with m1 macs. tested on m1 mac @danquack and x86 mac @drewmullen

**Comment by drewmullen** [comment]
LGTM and runs great. No strong opinion on including the linted files.

Only thing id like to get comment on before approving is `semgrep` METRICS. I just dont have the context to know if we should also pass `--metrics=off` in the makefile or not. any input, @ewbankkit 

```bash
make lint
==> Checking source code with golangci-lint...
==> Checking source code with terrafmt...
find ./internal/service -type f -name '*_test.go' \
    | sort -u \
    | grep -v efs/file_system_test.go \
    | grep -v kms/grant_test.go \
    | grep -v quicksight/user_test.go \
    | grep -v sns/platform_application_test.go \
    | xargs -I {} terrafmt fmt --check --fmtcompat {}
==> Checking source code with providerlint...
==> Checking source code with importlint...
==> Applying automatic website linter fixes...
==> Running Semgrep static analysis...
METRICS: Using configs from the Registry (like --config=p/ci) reports pseudonymous rule metrics to semgrep.dev.
To disable Registry rule metrics, use "--metrics=off".
Using configs only from local files (like --config=xyz.yml) does not enable metrics.

More information: https://semgrep.dev/docs/metrics

Running 27 rules...
ran 27 rules on 2003 files: 0 findings
```

**Comment by danquack** [comment]
@gdavison do you have any opinions as it relates to semgrep

**Pull request closed without merging**